pkg/network/http/model.go

```diff
@@ -38,27 +38,41 @@ func (k *httpBatchKey) Prepare(n httpNotification) {
 	k.page_num = C.uint(int(n.batch_idx) % HTTPBatchPages)
 }
 
-// Path returns the URL from the request fragment captured in eBPF with
-// GET variables excluded.
-// Example:
-// For a request fragment "GET /foo?var=bar HTTP/1.1", this method will return "/foo"
-func (tx *httpTX) Path(buffer []byte) []byte {
+func (tx *httpTX) validateFragment(outPath []byte) (bool, []byte) {
 	b := *(*[HTTPBufferSize]byte)(unsafe.Pointer(&tx.request_fragment))
 
-	var i, j int
+	// First, we'll check for the HTTP verb
+	var i int
 	for i = 0; i < len(b) && b[i] != ' '; i++ {
 	}
 
 	i++
 
-	for j = i; j < len(b) && b[j] != ' ' && b[j] != '?'; j++ {
+	if i >= len(b) || (b[i] != '/' && b[i] != '*') {
+		return false, nil
 	}
 
-	if i < j && j <= len(b) {
-		n := copy(buffer, b[i:j])
-		return buffer[:n]
+	if outPath != nil {
+		var j int
+		for j = i; j < len(b) && b[j] != ' ' && b[j] != '?'; j++ {
+		}
+
+		// no bound check necessary here as we know we at least have '/' character
+		n := copy(outPath, b[i:j])
+		return true, outPath[:n]
 	}
 
+	return true, nil
+}
+
+// Path returns the URL from the request fragment captured in eBPF with
+// GET variables excluded.
+// Example:
+// For a request fragment "GET /foo?var=bar HTTP/1.1", this method will return "/foo"
+func (tx *httpTX) Path(buffer []byte) []byte {
+	if ok, match := tx.validateFragment(buffer); ok {
```

Comment:
Returning `nil` when path is not valid is already what we do. I introduced the extra method because of the `IsValid` and `Path` pattern I had, but not super necessary now. Although IMO doesn't hurt, and I think it makes it easier to expand validation in the future given it's not in the `Path` method.

Reply:
Is there a real world scenario in which we would handle `true,nil` and `false,nil` differently? (or conversely `true,non-nil`, `false,non-nil`). If not then I'd definitely remove it as it only adds confusion